src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -0,0 +1,90 @@
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+import org.broadinstitute.dsde.firecloud.service.TrialService.{EnableUser, EnrollUser, TerminateUser}
+import spray.http.StatusCodes._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object TrialService {
+  sealed trait TrialServiceMessage
+  case class EnableUser(userInfo:UserInfo) extends TrialServiceMessage
+  case class EnrollUser(userInfo:UserInfo) extends TrialServiceMessage
+  case class TerminateUser(userInfo:UserInfo) extends TrialServiceMessage
+
+  def props(service: () => TrialService): Props = {
+    Props(service())
+  }
+
+  def constructor(app: Application)()(implicit executionContext: ExecutionContext) =
+    new TrialService(app.samDAO, app.thurloeDAO)
+}
+
+class TrialService
+  (val samDao: SamDAO, val thurloeDao: ThurloeDAO)
+  (implicit protected val executionContext: ExecutionContext)
+  extends Actor with LazyLogging {
+
+  override def receive = {
+    case EnableUser(userInfo) => enableUser(userInfo) pipeTo sender
+    case EnrollUser(userInfo) => enrollUser(userInfo) pipeTo sender
+    case TerminateUser(userInfo) => terminateUser(userInfo) pipeTo sender
+  }
+
+  // TODO: implement fully! Check that the user does not already have a state, before overwriting.
+  // this method exists solely for developer-testing purposes right now.
+  private def enableUser(userInfo: UserInfo): Future[PerRequestMessage] = {
+    // build the state that we want to persist to indicate the user is enabled
+    val now = System.currentTimeMillis
+    val enabledStatus = UserTrialStatus(userInfo.id, Some(TrialStates.Enabled), now, 0, 0, 0)
+    thurloeDao.saveTrialStatus(userInfo, enabledStatus) map { _ =>
+      RequestComplete(spray.http.StatusCodes.OK)
+    }
+  }
+
+  private def enrollUser(userInfo:UserInfo): Future[PerRequestMessage] = {
+    // get user's trial status, then check the current state
+    thurloeDao.getTrialStatus(userInfo) flatMap { userTrialStatus =>
+      userTrialStatus match {
+        // can't determine the user's trial status; don't enroll
+        case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial."))
+        case Some(status) =>
+          status.currentState match {
+            // user already enrolled; don't re-enroll
+            case Some(TrialStates.Enrolled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are already enrolled in a free trial."))
+            // user enabled (eligible) for trial, enroll!
+            case Some(TrialStates.Enabled) => {
+              // build the new state that we want to persist to indicate the user is enrolled
+              val now = System.currentTimeMillis
+              val enrolledStatus = status.copy(
+                currentState = Some(TrialStates.Enrolled),
+                enrolledDate = now,
+                expirationDate = now + (60 * 24 * 60 * 60 * 1000) // add 60 days from now
```

Comment:
Should `60` (days) be set in a config, or at least somewhere higher-level? 

Using a date/time library may also save the conversion work and make it less error-prone.

Reply:
yes, the 60-days should be stored in config! That's my other comments ... I'm trying to batch up the config updates because they're a pain.